proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java

```diff
@@ -424,16 +424,21 @@ public static VelocityConfiguration read(Path path) throws IOException {
     CommentedFileConfig defaultConfig = CommentedFileConfig.of(tmpFile, TomlFormat.instance());
     defaultConfig.load();
 
-    // Handle any cases where the config needs to be saved again
+    // Retrieve the forwarding secret. First, from environment variable, then from config.
     byte[] forwardingSecret;
-    String forwardingSecretString = config.get("forwarding-secret");
+    String forwardingSecretString = System.getenv("VELOCITY_FORWARDING_SECRET");
```

Comment:
```suggestion
    String forwardingSecretString = System.getenv("VELOCITY_FORWARDING_SECRET", config.get("forwarding-secret"));
```
This would allow you to remove the the first if block (+ the config fetch in line 431 ofc)

Reply:
`System#getenv` does not provide a "default" parameter.

Reply:
Whoops, my bad, confused it with system properties. Here's what I should've suggested :D
```suggestion
    String forwardingSecretString = System.getenv().getOrDefault("VELOCITY_FORWARDING_SECRET", config.get("forwarding-secret")
```